### PR TITLE
feature/emerald_components

### PR DESCRIPTION
**Pull request by ronalfabra**

### CODE REVIEW

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormContainer.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import SwiftUI
+
+@available(iOS 13.0.0, *)
+struct EmeraldTextFieldFormContainer<Content>: View where Content: View {
+    private let text: Binding<String>
+    private let hint: String?
+    private let placeholder: String?
+    private let errorText: String?
+
+    @State var isEditing: Bool
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormContainer.swift:18]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//
+
+import Foundation
+
+final class EmeraldTextFieldFormDateCoordinator: TextFormatter {
+    private var formatValidator = FormatValidator()
+    private let dateFormat: EmeraldIOS.TextFormat
+    private lazy var dateFormatter = DateFormatter()
+    
+    var minimumDate: Date?
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:16]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    var maximumDate: Date?
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:17]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    
+    init(dateFormat: TextFormat) {
+        self.dateFormat = dateFormat
+        setupDefaultDateFormat()
+    }
+    
+    func getFormat() -> TextFormat {
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:24]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    
+    func apply(to date: String) -> String? {
+        return try? apply(format: dateFormat, to: date)
+    }
+    
+    func remove(to date: String) -> String? {
+        return try? remove(format: dateFormat, to: date)
+    }
+    
+    func fillYear(date: String?) -> String {
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:36]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        formatValidator.fillYear(date: date)
+    }
+    
+    func getFormatDate(_ selectedDate: Date) -> String {
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:40]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        dateFormatter.string(from: selectedDate)
+    }
+    
+    func getDate(from string: String) -> Date? {
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:44]: It should be private

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = getDateFormat()
+        dateFormatter.locale = Locale(identifier: Constants.Values.localIdentifier)
+        return dateFormatter.date(from: string)
+    }
+    
+    func formatText(_ text: String) -> String {
```
Comment (RonaldRuizCondor) [inline EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:51]: It should be private

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 9]
```diff
@@ -8,6 +8,24 @@
...
+        name: String) {
+        self.id = id
+        self.name = name
+    }
+    
+    func getSelectableText() -> String {
+        name
+    }
+    
+
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:26]: Delete white space

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 10]
```diff
@@ -41,6 +59,18 @@ struct SwiftUiView: View {
     @State(initialValue: nil) var errorText: String?
     @State(initialValue: nil) var errorTextNew: String?
     @State var selected: String?
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:43]: It should be private

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 11]
```diff
@@ -41,6 +59,18 @@ struct SwiftUiView: View {
...
+    
+    @State(initialValue: "") var textV2: String
+    @State(initialValue: nil) var errorTextV2: String?
+    @State(initialValue: "") var textV3: String
+    @State(initialValue: nil) var errorTextV3: String?
+    
+    @State var pickerData: [Selectable] = [DataSelectable(id: "1", name: "name 1"),
+                                           DataSelectable(id: "2", name: "name 2")]
+    @State var itemPickerSelected: Int?
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:70]: It should be private

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 12]
```diff
@@ -41,6 +59,18 @@ struct SwiftUiView: View {
...
+    @State(initialValue: "") var textV4: String
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:71]: It should be private

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 13]
```diff
@@ -529,13 +595,13 @@ let radioGroups: [radioButtonModel] = [
     radioButtonModel(id: "2", title: "prueba2", requiredExplanation: true),
     radioButtonModel(id: "3", title: "prueba3", requiredExplanation: false),
     radioButtonModel(id: "4", title: "prueba4", requiredExplanation: true)
-  
+
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:598]: Delete white specs

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 14]
```diff
@@ -529,13 +595,13 @@ let radioGroups: [radioButtonModel] = [
     radioButtonModel(id: "2", title: "prueba2", requiredExplanation: true),
     radioButtonModel(id: "3", title: "prueba3", requiredExplanation: false),
     radioButtonModel(id: "4", title: "prueba4", requiredExplanation: true)
-  
+    
 ]
 
 struct radioButtonModel {
-
+
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:602]: Delete white space

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 15]
```diff
@@ -529,13 +595,13 @@ let radioGroups: [radioButtonModel] = [
...
 ]
 
 struct radioButtonModel {
-
+    
     let id: String
     let title: String
     var requiredExplanation: Bool
-
+
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:606]: Delete white space

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift [hunk 16]
```diff
@@ -122,40 +152,40 @@ struct SwiftUiView: View {
...
-            EmeraldSwiftUIButton(
-                buttonName: _buttonName,
-                isEnabled: .constant(true),
-                isHighlighted: .constant(false),
-                themeStyle: .primaryLarge) {}
+                EmeraldSwiftUIButton(
+                    buttonName: _buttonName,
+                    isEnabled: .constant(true),
+                    isHighlighted: .constant(false),
+                    themeStyle: .primary) {
```
Comment (RonaldRuizCondor) [inline ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:159]: The closure should not be mandatory

**Review by RonaldRuizCondor — CHANGES_REQUESTED** [review]

**Review by jmenco — DISMISSED** [review]

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 16 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:159]: this was developed in other sprint (other task), I don't modified this class for my scope.

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 15 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:606]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 14 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:602]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 13 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:598]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 12 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:71]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 11 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:70]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 10 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:43]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 9 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:26]: done

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 8 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:51]: it was added in protocol, done.

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 7 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:44]: it was added in protocol, done.

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 6 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:40]: it was added in protocol, done.

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 5 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:36]: it was added in protocol, done.

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 4 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:24]: it was added in protocol, done.

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 3 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:17]: done

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 2 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:16]: done

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormContainer.swift (hunk 1 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormContainer.swift:18]: done

File: ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift (hunk 14 above)
Comment (ronalfabra) [reply ProofOfConcept/ProofOfConcept/SwiftUi/SwiftUiView.swift:602]: done

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 5 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:36]: it was added in protocol, done.

File: EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift (hunk 4 above)
Comment (ronalfabra) [reply EmeraldIOS/Components/SwiftUi/TextField/TextFieldForm/EmeraldTextFieldFormDate/EmeraldTextFieldFormDateCoordinator.swift:24]: it was added in protocol, done.

**Review by RonaldRuizCondor — APPROVED** [review]

**Review by jmenco — APPROVED** [review]

**Pull request merged**